bundles/AdminBundle/Controller/Admin/Asset/AssetController.php

```diff
@@ -520,6 +520,15 @@ protected function addAsset(Request $request, Config $config)
             throw new \Exception('Something went wrong, please check upload_max_filesize and post_max_size in your php.ini as well as the write permissions of your temporary directories.');
         }
 
+        if($request->get('accept')) {
+            $mimetype = MimeTypes::getDefault()->guessMimeType($sourcePath);
+            $accept = $request->get('accept');
+            $acceptedMimeTypes = explode(',', $accept);
+            if(!$this->_assetService->validateMimeType($mimetype, $acceptedMimeTypes)) {
```

Comment:
@mattamon could you please adapt it as per https://github.com/pimcore/pimcore/pull/13464#issuecomment-1293313339? so we should just check the type here instead of validating mime type. I would get rid of accept property from upload dialog and just compare the type with the value from `getTypeFromMimeMapping()` on server side.

Reply:
I think the server side validation only is a bit lacking. 
As mention in the issue, it is possible to add e.g. a .docx to an image tag with no validation. The file is uploaded but will never be displayed in the object, because of the wrong type.

On the other hand, what do I compare the new file type to, since I the type is not specified anywhere?

If I upload a .jpg first and then a .docx, I don't find the file (the extension is part of the filename -> `Asset\Service::pathExists($parentAsset->getRealFullPath().'/'.$filename)`) and a new one will be created, or did I miss something?

Reply:
@mattamon please find my answers below:
> The file is uploaded but will never be displayed in the object, because of the wrong type.

yes, we should refactor the calls here to cleanup the uploaded file even in case of exceptions https://github.com/pimcore/pimcore/blob/8d527233a22fe230d57d84fa6c7d7ae3e793cfa8/bundles/AdminBundle/src/Controller/Admin/Asset/AssetController.php#L371 

> On the other hand, what do I compare the new file type to, since I the type is not specified anywhere?

we can still pass this property (with correct value) to the server for comparison https://github.com/pimcore/pimcore/pull/13464/files#diff-fe453464481618fcfe5a3be91df93fed62f7f00b8053f32ebdd4c97dfdfbf6b2R19

> If I upload a .jpg first and then a .docx, I don't find the file (the extension is part of the filename -> Asset\Service::pathExists($parentAsset->getRealFullPath().'/'.$filename)) and a new one will be created, or did I miss something?

This is quite an edge case, if you try to upload files with same names and different extensions so i would not consider it.

The reason we'd like to just compare the type as you won't be able to validate all image types with one group (e.g. `image/*`) so it is better to use `getTypeFromMimeMapping()` for comparison.

Reply:
@mattamon are you on it?